proto/substrait/algebra.proto

```diff
@@ -364,6 +364,47 @@ message Rel {
   }
 }
 
+// The operator that modifies the schema/content of a database
+message WriteRel {
```

Comment:
This seems to be the analogue / opposite of a "named table read".  In Arrow we also have a write rel but it is the analogue / opposite of a "local files read".  We take in options like "destination directory", "file format", "partitioning", etc.

It would be nice to structure things in a way that the ReadRel mirrors the WriteRel which, in this case, I think would mean introducing a `WriteType` and having this content (`table`, `base_schema`, `defaults`) be inside a `NamedTableWrite` message or something similar.

Reply:
@westonpace 

Regarding `WriteType` I like, let me and @jcamachor  try. 

As for DDL, I am not worried about the "read" path, as this is handled by most (all?) RDBMS by simple using `SELECT` on a special DB that is about metadata of the system (usually called `information_schema`), so things would look like:

```
SELECT table_name, table_schema, table_type
FROM information_schema.tables
```

(And there are other tables for views, indexes, constraints, columns, etc..)

Reply:
If you like this version, I can propagate to website etc. We will also likely do in a separate patch the isthmus changes needed to test all this out.

Reply:
> **usually** called `information_schema`

If one of the goals of Substrait is to try to remedy the current situation where each system has its own flavor of SQL, rather than just being binary SQL, I don't think that should be the way to go. It also doesn't extend to metadata queries for file data sources.

I'm trying not to get too involved in these additions because with my background I feel like an outsider looking in, but as that outsider, generally I feel like lots of things are currently being specified based on how SQL has always pragmatically done things, rather than based on what makes sense. For example, why should something like a `drop table` or `insert into` return a table? Virtually every other language has a statement-vs-expression-like concept to allow for things that don't return anything. Elsewhere I joked about `SELECT * FROM (DROP TABLE *)` to masquerade a malicious plan as something immutable; I don't know if SQL allows this but if these DDL statements are modeled as regular 1-input 1-output relations I see no reason why Substrait wouldn't. I personally liked the original proposal that extended the PlanRel oneof a lot more than using WriteRel for this reason. Just my two cents I guess.

Reply:
I wonder whether we could "extend to metadata queries for file data sources" by introducing an operator that extract metadata from a file as a relation, and then allow the standard `SELECT * FROM information_schema.tables` type tricks, which I find very pleasant in  RDBMSs as you get to use all the tools and language expressivity you have for data also for metadata. 

`drop table` (and CREATE/ALTER for table and VIEWS) do not return a rel (in the latest patch), but `INSERT/DELETE/UPDATE` do. Why does SQL support that I am not sure, but poking around I have seen folks in blogs/discussions do interesting things to compactly execute a statement and verify properties about what happened in the DBMS, avoiding odd workaround like selecting into temporary tables, executing and then dropping temp tables, etc. Intuitively it helps keep the language as "close" as possible, which I think is very convenient in many ways (I think @jacques-n  hints at that below).